### DISCUSSION TITLE
Sqlite ON DELETE CASCADE

### DISCUSSION THREAD

**Discussion by Christian-Fenzl** [Mimetis/Dotmim.Sync#1219] [Q&A]
Hi,

my server database is MS SQL SERVER (created by EF Core Migrations) and the client database is Sqlite. The Sqlite database is completely created by the first sync with Dotmim.Sync.

The server database contains some parent/child relationships with enabled ON DELETE CASCADE on their foreign keys. This ON DELETE CASCADE is missing in the created Sqlite database, even with enabled "Foreign Keys=True;" in the connection string for the Sqlite database.

Is that by design to not include the ON DELETE CASCADE in the synched client database? I read some advise to better not use it with Dotmim.Sync but nothing really clear.

So is it better to not rely on this option and delete the childs records explicitly by myself?
Or how can I enable the creation during the first sync?

**Comment by Mimetis** [comment]
> Is that by design to not include the ON DELETE CASCADE in the synched client database

Yes, DMS is not a database generator, so the client sql generation is pretty limited.

And as you say, the DELETE CASCADE is not tested (and will not) 
So I guess disabling it could be a good idea :)
